packages/next/src/client/components/router-reducer/prefetch-cache-utils.ts

```diff
@@ -133,7 +140,43 @@ function prefixExistingPrefetchCacheEntry({
 }
 
 /**
- * Creates a prefetch entry for data that has not been resolved. This will add the prefetch request to a promise queue.
+ * Use to seed the prefetch cache with data that has already been fetched.
+ */
+export function createPrefetchEntry({
```

Comment:
Don't love the naming here, since it's really only used for a single use case. Should be more specific, like `createPrefetchCacheEntryForInitialPageLoad` or whatever. If the logic of the prefetch cache weren't so subtle it would maybe be better to inline it into app-router but I like that you've kept it here, since it likely needs to change whenever the prefetch cache implementation changes.